sqlccl: correctly ignore dropped tables during backup

madelynnblue opened the pull request:
Dropping databases and tables failed in different ways, hence the
separate tests, but the fix is the same.

Fixes #19020

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/19032)
<!-- Reviewable:end -->

benesch commented on pkg/ccl/sqlccl/backup_test.go:
```diff
@@ -2008,3 +2008,62 @@ func TestPointInTimeRecovery(t *testing.T) {
+
+	sqlDB.Exec("CREATE DATABASE data2")
+
+	if err := verifyBackupRestoreStatementResult(
```
I don't think `verifyBackupRestoreStatementResult` is the right verification routine to use. That function exists purely to support `TestBackupRestoreStatementResult`, which cross-checks `crdb_internal.jobs` against the information returned from the `BACKUP` or `RESTORE` statement.

It think you probably want to do something like what `TestRestoreInto` does, which verifies a `SELECT * FROM data2.foo` against `SELECT * FROM data.foo`.

mjibson replied:
Ah. Done.

madelynnblue committed:
sqlccl: correctly ignore dropped tables during backup

Dropping databases and tables failed in different ways, hence the
separate tests, but the fix is the same.

Fixes #19020
